zulip/zulip/__init__.py

```diff
@@ -1539,6 +1539,23 @@ def get_storage(self, request: Optional[Dict[str, Any]] = None) -> Dict[str, Any
             request=request,
         )
 
+    def remove_storage(self, keys: List[str] = []) -> Dict[str, str]:
```

Comment:
I disagree with @LoopThrough-i-j: I think this default of `[]` is wrong and should be `None`, because the behavior of `remove_storage()` with no arguments should be to remove all keys, which is different than removing no keys. See https://zulip.readthedocs.io/en/latest/testing/mypy.html#use-optional-and-none-correctly.

It happens that our `DELETE /api/v1/bot_storage` with `keys = []` also has the effect of removing all keys, but that’s a side effect of its misuse of falsiness in `keys = keys or get_keys_in_bot_storage(user_profile)`, and I think that should be fixed.

Reply:
Okay then keys should be `keys: Optional[List[str]] = None`, and we should put a proper check-in the actual API. 

This line `keys = keys or get_keys_in_bot_storage(user_profile)`, bothered me about the proper usage.

Reply:
`[]` this should ideally not delete anything then.

Reply:
I'm very sorry for the inconvenience, but I accidentally forced pushed a wrong commit, which closed this pull request, and I can not reopen it to redo my errors, it was an honest mistake from my end, and I believe I can not reopen it, because it got merged?, I'm completely lost at the moment.

Reply:
@R0cker77 Yeah, force-pushing the PR to an ancestor of the target branch had the effect of closing it in such a way that it can’t be reopened. You’ll need to open a new PR.

Reply:
Yeah, I'm going to make a new pr with `keys: Optional[List[str]] = None`.
And again I'm sorry about this mess.